kale/worker.py

```diff
@@ -222,7 +227,11 @@ def _cleanup_worker(self, signum, frame):
         """
         logger.info('Process sent signal %d. Cleaning up tasks...' % signum)
 
+        self._cleanup_lock.acquire()
+        self._terminated = True
```

Comment:
+1. I'd also be inclined to push the lock down into the self._release_batch().  I don't think self._terminated = True needs to be protected by the lock.